code/obj/machinery/computer/security/riot.dm

```diff
@@ -1,15 +1,23 @@
 /obj/machinery/computer/riotgear
 	name = "Armory Authorization"
+	desc = "Use this computer to authorize security access to the Armory. You need an ID with security access to do so."
 	icon_state = "drawbr"
 	density = 0
 	glow_in_dark_screen = TRUE
 	var/auth_need = 3.0
-	var/list/authorized
-	var/list/authorized_registered = null
 	var/net_id = null
 	var/control_frequency = "1461"
 	var/radiorange = 3
-	desc = "Use this computer to authorize security access to the Armory. You need an ID with security access to do so."
+
+	var/list/authorisations = list()
+	var/hasID = FALSE
```

Comment:
A bit confused why you're storing the auth state on the obj since it's always going to be per player, so surely all storing it will do is cause bugs?

Reply:
I do not quite understand what you mean by that, do you mean `var/list/authorisations` or the several other vars?

Reply:
The several other vars that make up the current authorization state, since multiple copies of the interface can exist at once storing these on the object and overwriting them every time someone opens/interacts with the interface is going to lead to them not matching the current user's authorization state surely?

Reply:
It is, yes, I should have caught that.

> You're entirely right, yes... I'm not quite sure how to prevent that; what would you recommend for storing that data per player?